File: DiceRange.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TableDieRangeCalculator
{
    public class DiceRange<TEntry>
    {
        public static string Nil = "-";

        protected internal DiceRange(TEntry entry)
        {
            Entry = entry;
        }

        public TEntry Entry { get; }
        public int MinRoll { get; set; }
        public int MaxRoll { get; set; }
        public decimal AssignedWeight { get; set; }
        public decimal TargetWeight { get; set; }
        public virtual string WriteRange()
        {
            if (MinRoll == 0 && MaxRoll == 0)
            {
                return Nil;
            }
            else if (MinRoll == MaxRoll)
            {
                return MinRoll.ToString();
            }
            else
            {
                return $"{MinRoll}-{MaxRoll}";
            }
        }
    }
}

File: node/src/diceRange.js
'use strict';

const { Decimal } = require('./decimal');

class DiceRange {
  constructor(entry) {
    this.entry = entry;
    this.minRoll = 0;
    this.maxRoll = 0;
    this.assignedWeight = Decimal.ZERO;
    this.targetWeight = Decimal.ZERO;
  }

  writeRange() {
    if (this.minRoll === 0 && this.maxRoll === 0) return DiceRange.Nil;
    if (this.minRoll === this.maxRoll) return String(this.minRoll);
    return `${this.minRoll}-${this.maxRoll}`;
  }
}

DiceRange.Nil = '-';

function pad2(value) {
  return String(value).padStart(2, '0');
}

class PercentileDiceRange extends DiceRange {
  writeRange() {
    if (this.minRoll === 0 && this.maxRoll === 0) return super.writeRange();
    if (this.minRoll === this.maxRoll) {
      return this.minRoll === 100 ? PercentileDiceRange.DoubleZero : pad2(this.minRoll);
    }
    if (this.maxRoll === 100) return `${pad2(this.minRoll)}-${PercentileDiceRange.DoubleZero}`;
    return `${pad2(this.minRoll)}-${pad2(this.maxRoll)}`;
  }
}

PercentileDiceRange.DoubleZero = '00';

module.exports = { DiceRange, PercentileDiceRange };
